docs: improve documentation (#5883)
DimensionDev/Maskbook: docs/bounty-development-guide.md

# Bounty Development Guide

Hi, Awesome people! Welcome to start a bounty task on Mask Network.

## Tech Reqirements

> Mask Network extension is mainly written in TypeScript. The UI part is building with React upon [MUI](https://mui.com/getting-started/installation/) framework. Also, we write [CSS-in-JS](css-in-js.md). Use webpack to build the project. Besides that, we prefer widely adopting tech solutions that includes:

Reviewer: ```suggestion
Mask Network extension is written in TypeScript. The UI is written by React and [@mui](https://mui.com/) framework. We write CSS in [CSS-in-JS](css-in-js.md) style.

We prefer widely adopting tech solutions that include:
```

- [Web3.js](https://web3js.readthedocs.io/) Ethereum JavaScript API
- [react-use](https://streamich.github.io/react-use/) React Hooks — 👍
- [bignumber.js](https://mikemcl.github.io/bignumber.js/) A JavaScript library for arbitrary-precision arithmetic.
- [lodash](https://lodash.com/docs/) A modern JavaScript utility library delivering modularity, performance & extras.
- [urlcat](https://urlcat.dev/) A URL builder library for JavaScript.

> If your bounty task is related to another project, it could have extra requirements that need to consider.

> The codebase is totally open sourced. Anyone can review it. If you are familiar with these libraries mentioned above, it will take less friction for you to get start.

Reviewer: ```suggestion
If you are familiar with these libraries mentioned above, it will take less effort for you to get started.
The codebase is open-sourced under the AGPLv3 license.
```

## General Content

> Firstly, you can clone the repository and [set up the development environment](setup.md). The codebase is easy to understand, although with a huge size. It divides into many packages. Each package serves a specific purpose, as its name says. Let's take a quick tour.

Reviewer: ```suggestion
After cloning the repository and [set up the development environment](setup.md). The codebase is constructed as a monorepo with many internal packages. Each package serves a specific purpose. Let's take a quick tour.
```

### Core Packages

> - `packages/mask` The core part of the extension which adapts multiple social networks into the system, keeps the user's data safty, implements encrpytion & decryption protocols, and hosts a plugin system.

Reviewer: ```suggestion
- `packages/mask` The main extension which has multiple websites supports, keeps the user's data safe and hosts a plugin system.
- `packages/encryption` The encryption & decryption of mask network.
```
